docs/modules/spark-k8s/pages/usage-guide/configuration-environment-overrides.adoc

```diff
@@ -0,0 +1,132 @@
+= Configuration & Environment Overrides
+
+The cluster definition also supports overriding configuration properties and environment variables, either per role or per role group, where the more specific override (role group) has precedence over the less specific one (role).
+
+IMPORTANT: Overriding operator-set properties (such as the ports) can interfere with the operator and can lead to problems.
+
+
+== Configuration Properties
+
+For a role or role group, at the same level of `config`, you can specify `configOverrides` for the following files:
+
+* `spark-env.sh`
+* `security.properties`
+
+NOTE: `spark-defaults.conf` is not required here, because the properties defined in {crd-docs}/spark.stackable.tech/sparkhistoryserver/v1alpha1/#spec-sparkConf[`sparkConf` (SparkHistoryServer)] and {crd-docs}/spark.stackable.tech/sparkapplication/v1alpha1/#spec-sparkConf[`sparkConf` (SparkApplication)] are already added to this file.
+
+For example, if you want to set the `networkaddress.cache.ttl`, it can be configured in the SparkHistoryServer resource like so:
+
+[source,yaml]
+----
+nodes:
+  roleGroups:
+    default:
+      configOverrides:
+        security.properties:
+          networkaddress.cache.ttl: "30"
+      replicas: 1
+----
+
+Just as for the `config`, it is possible to specify this at the role level as well:
+
+[source,yaml]
+----
+nodes:
+  configOverrides:
+    security.properties:
+      networkaddress.cache.ttl: "30"
+  roleGroups:
+    default:
+      replicas: 1
+----
+
+All override property values must be strings.
+
+The same applies to the `job`, `driver` and `executor` roles of the SparkApplication.
+
+=== The spark-env.sh file
+
+The `spark-env.sh` file is used to set environment variables.
+Usually, environment variables are configured in `envOverrides`, but `envOverrides` only allow to set static values.
```

Comment:
```suggestion
Usually, environment variables are configured in `envOverrides`, but `envOverrides` only allows static values to be set.
```

Reply:
Aren't `envOverrides` plural, even if it is only a keyword?

Reply:
Yes, probably :) Then it would be "...only allow the setting of static values" or "...only allow static values to be set"

Reply:
4deda3baf5f4f73f065785d4e4c268245c2836f0